todo.ts

```diff
@@ -0,0 +1,64 @@
+interface ITask {
+    name: string,
+    date: Date,
+    description: string,
+    completed: boolean
+}
+
+class Task implements ITask {
+    constructor(public name:string, public date: Date, public description: string, public completed: boolean){
+    }
+    finish() {
+        this.completed = true;
+    }
+    changeProperty(propName: string, newValue: string) {
+        this[propName] = newValue;
+    }
+
+}
+
+class ToDoList {
+    public tasks: Task[]=[];
+    addTask(newTask: Task){
+        this.tasks.push(newTask);
+    }
+    finishTask(index: number) {
+        this.tasks[index].finish();
+    }
+    allTasks() {
+        console.table(this.tasks);
+    }
+    changeProp(index: number, propName: string, newValue: string) {
+        for (let props in this.tasks[index]) {
+            if (props == propName){
+                this.tasks[index].changeProperty(propName, newValue)
+                return;
+            }
+        }
+        console.log(`${propName} does not exist`);
+    }
+    unfinishedTasks() {
+        const notFinished = this.tasks.filter( el => el.completed === false);
+        console.table(notFinished);
+    }
+    removeUnfinishedTask(index: number) {
+        if (this.tasks[index].completed === false) {
```

Comment:
Grija la verificarea asta `this.tasks[index]` poate fi undefined, iar accesarea proprietatii completed pe undefined va genera eroare :)

Reply:
Multumesc pentru comentarii. Am inteles problemele si le-am corectat.